packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -33,39 +33,97 @@ import {
 export async function runBrokenLinks(
   options: BrokenLinksResultV1_BrokenLinkCheckerOptions
 ): Promise<SyntheticResult> {
-  // START - to resolve warnings while under development
-  checkStatusPassing({ status_value: 200 } as ResponseStatusCode, 200);
-
-  const browser = await puppeteer.launch({ headless: 'new' });
-  const page = await browser.newPage();
-  await checkLink(
-    page,
-    {} as LinkIntermediate,
-    {} as BrokenLinksResultV1_BrokenLinkCheckerOptions
-  );
-  // END - to resolve warnings  while under development
-
   // options object modified directly
   setDefaultOptions(options);
 
-  // PSEUDOCODE
-
-  // create puppeteer.Browser
-  // create puppeteer.Page & navigate to origin_url, w/ origin specific settings
+  // create Browser & origin page then navigate to origin_url, w/ origin
+  // specific settings
+  const browser = await puppeteer.launch({ headless: 'new' });
+  const origin_page = await browser.newPage();
+  if (options.wait_for_selector)
+    await origin_page.waitForSelector(options.wait_for_selector);
 
   // scrape origin_url for all links
-  // (shuffle links if necessary and) truncate at link_limit
+  const retrieved_links: LinkIntermediate[] = await retrieveLinksFromPage(
+    origin_page,
+    options.query_selector_all,
+    options.get_attributes
+  );
+
+  // TODO:  (shuffle links if necessary and) truncate at link_limit
 
+  // TODO
   // create new page to be used for all scraped links
   // navigate to each link - LOOP:
   //          each call to `checkLinks(...)` will return a `SyntheticLinkResult`
   //          Object added to an array of `followed_links`
 
   // returned a SyntheticResult with `options`, `followed_links` &
   // runtimeMetadata
+
+  // Resolving warnings - START IGNORE (will be removed)
+  await checkLink(
+    origin_page,
+    {} as LinkIntermediate,
+    {} as BrokenLinksResultV1_BrokenLinkCheckerOptions
+  );
+  retrieved_links;
+  // Resolve warnings - END IGNORE (will be removed)
   return {} as SyntheticResult;
 }
 
+/**
+ * Retrieves all links on the page using Puppeteer, handling relative and
+ * protocol-relative links and filtering for HTTP/HTTPS links.
+ *
+ * @param page - The Puppeteer page instance to retrieve the links from.
+ * @param query_selector_all - The CSS selector to identify link elements.
+ * @param get_attributes - An array of attribute names to retrieve from the link elements.
+ * @returns An array of LinkIntermediate objects representing the links found.
+ */
+export async function retrieveLinksFromPage(
```

Comment:
Since the index.ts file exports all functions from this file, retrieveLinksFromPage essentially becomes a part of the public contract. Would this be better suited in a util file of some sort?

Reply:
Logically and for readability I think it makes sense to keep this function in this file.

Would it be fine if I updated index.ts to be:
```
export {
  runBrokenLinks,
  BrokenLinkCheckerOptions,
  PerLinkOption,
  StatusClass,
  LinkOrder,
} from './broken_links';
```

Reply:
Yeah that works